pages/api/teams/[slug]/payments/create-portal-link.ts

```diff
@@ -0,0 +1,41 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import { getSession } from '@/lib/session';
+import { throwIfNoTeamAccess } from 'models/team';
+import { stripe, getURL, getStripeCustomerId } from '@/lib/stripe';
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    switch (req.method) {
+      case 'POST':
+        await handlePOST(req, res);
+        break;
+      default:
+        res.setHeader('Allow', 'POST');
+        res.status(405).json({
+          error: { message: `Method ${req.method} Not Allowed` },
+        });
+    }
+  } catch (error: any) {
+    const message = error.message || 'Something went wrong';
+    const status = error.status || 500;
+
+    res.status(status).json({ error: { message } });
+  }
+}
+
+const handlePOST = async (req: NextApiRequest, res: NextApiResponse) => {
+  const teamMember = await throwIfNoTeamAccess(req, res);
+  const session = await getSession(req, res);
+  const customerId = await getStripeCustomerId(teamMember, session);
+
+  const { url } = await stripe.billingPortal.sessions.create({
+    customer: customerId,
+    return_url: `${getURL()}teams/${teamMember.team.slug}/payments`,
+  });
+
+  res.json({ url });
```

Comment:
Can we wrap this within data object? `{ data: { url } }` to have consistent response format.